.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,41 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'r**'
+    types: [ labeled, synchronize ]
```

Comment:
`synchronize` makes it so that our GPU CI is run every time a new commit is added (provided the `gpuci` label is still there). This makes it more convenient for us, however it may cause security risks, and NeMo does not use it.

It probably makes sense for us not to use `synchronize`, but I wanted to include it here for consideration.